include/zephyr/drivers/serial/uart_async_to_irq.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SERIAL_UART_ASYNC_TO_IRQ_H_
+#define ZEPHYR_DRIVERS_SERIAL_UART_ASYNC_TO_IRQ_H_
+
+#include <zephyr/drivers/uart.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/serial/uart_async_rx.h>
+
+/**
+ * @brief UART Asynchronous to Interrupt driven API adaptation layer
+ * @ingroup uart_interface
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Forward declarations. */
+
+/** @brief Data structure used by the adaptation layer.
+ *
+ * This data must be the first element of the UART device data structure.
```

Comment:
A pointer to this data must be that first element. Same for the config.

Reply:
Added. Also added static asserts in the shim to check that.